### ISSUE TITLE
Is it possible to use on a machine without AVX2?

### ISSUE THREAD

**Issue by kevingnet** [Blosc/c-blosc#177]
This may have already been explained in the doc, however I was unable to find any information.

I'm trying to use the library in a Xeon processor which is not the latest and does not have AVX2. Is this supported?

Thanks.

**Comment by kevingnet** [comment]
Seems I was able to compile it. For some reason this didn't work before. I think I had an older version.

So it seems that when AVX2 is not available it will use SIMD2? Can it use AVX?

**Comment by jack-pappas** [comment]
Yes, you can use c-blosc on any x86 processor. The library includes hardware feature detection, so although it may have SSE2 and AVX2 support compiled in, the library will automatically choose the best shuffle implementation supported by the hardware it's running on. The "generic" implementation doesn't use any vectorization so should (at least in theory, if not in practice) run on any x86 processor.

c-blosc doesn't support AVX, only SSE2 and AVX2 (for x86 platforms). AVX doesn't have any instructions that'd be useful for c-blosc.
